Add summary of each backend to docs (#1385)
moj-analytical-services/splink: docs/topic_guides/backends.md

Splink is a Python library. It implements all data linking computations by generating SQL, and submitting the SQL statements to a backend of the user's chosing for execution.

The Splink code you write is almost identical between backends, so it's straightforward to migrate between backends. Often, it's a good idea to start working using DuckDB on a sample of data, because it will produce results very quickly. When you're comfortable with your model, you may wish to migrate to a big data backend to estimate/predict on the full dataset.

## Choosing a backend

### Considerations when choosing a SQL backend for Splink

> When choosing which backend to use when getting started with Splink, there are another factors to consider:

Reviewer: ```suggestion
When choosing which backend to use when getting started with Splink, there are a number of factors to consider:
```

> - the size of the datasets

Reviewer: ```suggestion
- the size of the dataset(s)
```

> - the amount of configuration required

Reviewer: This is ambiguous, but you may well address it below and I just haven't got there yet.

See the DuckDB [deduplication example notebook](../demos/example_deduplicate_50k_synthetic.ipynb) to get a better idea of how Splink works with DuckDB.

### :simple-apachespark: Spark

Spark is a system for distributed computing which is great for large datasets (10-100+ million records). It is more involved in terms of configuration, with more boilerplate code than the likes of DuckDB. Spark has complete coverage for the functions in the Splink [comparison libraries](../comparison_level_library.md) and, as a mainstay of the Splink development team, is actively maintained with features being added regularly.

If working with Databricks, the Spark backend is recommended, however as the Splink development team does not have access to a Databricks environment there will be instances where we will be unable to provide support.

See the Spark [deduplication example notebook](../demos/example_simple_pyspark.ipynb) to get a better idea of how Splink works with Spark.

### :simple-amazonaws: Athena

> Athena is a big data SQL backend provided on AWS which is great for large datasets (10+ million records). It requires access to a live AWS account and as a persistent database, requires some additional management of the tables created by Splink. Athena has reasonable, but not complete, coverage for the functions in the Splink [comparison libraries](../comparison_level_library.md), with gaps in string fuzzy matching functionality due to the lack of some string functions in Athena's underlying SQL engine, [Presto](https://prestodb.io/docs/current/). At this time, the Athena backend is not being actively used by the Splink development team so receives minimal levels of support.

Reviewer: ```suggestion
Athena is a big data SQL backend provided on AWS which is great for large datasets (10+ million records). It requires access to a live AWS account and as a persistent database, requires some additional management of the tables created by Splink. Athena has reasonable, but not complete, coverage for the functions in the Splink [comparison libraries](../comparison_level_library.md), with gaps in string fuzzy matching functionality due to the lack of some string functions in Athena's underlying SQL engine, [Presto](https://prestodb.io/docs/current/). At this time, the Athena backend is being used sparingly by the Splink development team so receives minimal levels of support.
```

or you could say "in niche use cases"

In addition, from a development perspective, the neccessity for an AWS connection makes testing Athena code more difficult, so there may be occassional bugs that would normally be caught by our testing framework.

See the Athena [deduplication example notebook](../demos/athena_deduplicate_50k_synthetic.ipynb) to get a better idea of how Splink works with Athena.

### :simple-sqlite: SQLite

> SQLite is similar to DuckDB in that it is, generally, more suited to smaller datasets. While not as performant as DuckDB, SQLite is simple to setup and can even be run directly in a Jupyter notebook. SQLite has reasonable, but not complete, coverage for the functions in the Splink [comparison libraries](../comparison_level_library.md), with gaps in array and date comparisons. String fuzzy matching, while not native to SQLite is available via python UDFs which has some [performance implications](#additional-information-for-specific-backends). SQLite is not actively been used by the Splink team so receives minimal levels of support.

Reviewer: The wording, "can even be run directly in a ...", makes it sound as if DuckDB can't do that here

Author: Updates
